### PR TITLE
test and publish

### PR DESCRIPTION
**Pull request by bhanuprakash-egov**

### CODE REVIEW

**Comment by bhanuprakash-egov** [comment]
@saiprakash-v @prasanna-egov

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        schema.status = "PUBLISHED"
+        axios.post('/vc-management/v1/schema', schema, {headers:{Authorization:{userToken}}})
+        .then(res =>res.data
+            ).catch(error => {
+                console.error(error);
+                throw error;
+            });
+    }
+    const [samplefile, setSamplefile] = useState(null);
+    // const schema = {
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:24]: remove the comments

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    //     "status": "DRAFT" 
+    // }
+    const requiredFeilds = (JSON.parse(schema.schema).definitions[schema.name].required).toString().split(",");
+    var formObj = {}; requiredFeilds.forEach(key => formObj[key] = "");
+    const [data, setData] = useState(formObj);
+    const previewReqBody = ({
+            credentialTemplate:JSON.parse(schema.schema)._osConfig.credentialTemplate,
+            // template:"Tenant/969b7d5c-4c2a-42a2-9d8f-5cd1348cf54d/templates/documents/527f7650-a8d0-42ae-a480-f01b9396bf72-trainingCert.html",
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:34]: same here

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import PrintIcon from '../../assets/img/print.svg';
+const axios = require('axios');
+
+const TestAndPublish = (props) => {
+    const schema = props.schema;
+    const {keycloak} = useKeycloak();
+    const setAsPublish = async () => {
+        const userToken = await getToken();
+        schema.status = "PUBLISHED"
+        axios.post('/vc-management/v1/schema', schema, {headers:{Authorization:{userToken}}})
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:16]: keep all the urls in one place

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      return axios.get(`${config.tokenEndPoint}/${userId}`).then(res =>
+      res.data.access_token.access_token
+    ).catch(error => {
+      console.error(error);
+      throw error;
+    });
+    };
+    const previewSchemaFunc = async () => {
+        const userToken = await getToken();
+        return axios.post(`/vc-management/v1/schema/preview`, previewReqBody,
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:54]: move urls to one place

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    }
+    const formInputHandler = (e) => {
+        setData({...data, [e.target.name]:e.target.value})
+    }
+  
+  return (
+    <div>
+        <div className='row mx-5 px-5'>
+        <div className='col-6'>
+            <h1>Test & Publish</h1>
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:82]: Add translations

**Review by saiprakash-v — CHANGES_REQUESTED** [review]

File: tenant-portal/src/components/TestAndPublish/TestAndPublish.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            // template:"Tenant/969b7d5c-4c2a-42a2-9d8f-5cd1348cf54d/templates/documents/527f7650-a8d0-42ae-a480-f01b9396bf72-trainingCert.html",
+            template: props.template,
+            data: data
+        })
+
+    const getUserId = async () => {    
+        const userInfo = await keycloak.loadUserInfo();
+        return userInfo.email;
+    }
+    const getToken = async () => {
```
Comment (saiprakash-v) [inline tenant-portal/src/components/TestAndPublish/TestAndPublish.js:43]: can these be moved to keycloak utils?

**Review by saiprakash-v — APPROVED** [review]

**Pull request merged**